src/Type/Symfony/ParameterDynamicReturnTypeExtension.php

```diff
@@ -101,13 +115,45 @@ private function getGetTypeFromMethodCall(
 		if ($parameterKey !== null) {
 			$parameter = $this->parameterMap->getParameter($parameterKey);
 			if ($parameter !== null) {
-				return $this->generalizeType($scope->getTypeFromValue($parameter->getValue()));
+				return $this->generalizeTypeFromValue($scope, $parameter->getValue());
 			}
 		}
 
 		return $returnType;
 	}
 
+	/**
+	 * @param Scope								 $scope
+	 * @param array<mixed>|bool|float|int|string $value
+	 */
+	private function generalizeTypeFromValue(Scope $scope, $value): Type
+	{
+		if (is_array($value) && $value !== []) {
+			return $this->generalizeType(
+				new ArrayType(
+					TypeCombinator::union(...array_map(function ($item) use ($scope): Type {
+						return $this->generalizeTypeFromValue($scope, $item);
+					}, array_keys($value))),
+					TypeCombinator::union(...array_map(function ($item) use ($scope): Type {
+						return $this->generalizeTypeFromValue($scope, $item);
+					}, array_values($value)))
+				)
+			);
+		}
+
+		if (
+			is_string($value)
+			&& preg_match('/%env\((.*)\:.*\)%/U', $value, $matches) === 1
+			&& strlen($matches[0]) === strlen($value)
+		) {
+			$providedTypes = EnvVarProcessor::getProvidedTypes();
```

Comment:
This introduces a hard dependency on `symfony/dependency-injection`. I get the idea about not duplicating Symfony's internal code here, but if we want to do it like that, we need to use something like `call_user_func(['\Symfony\Component\DependencyInjection\EnvVarProcessor', 'getProvidedTypes'])`.

Reply:
I got the opposite review here: https://github.com/phpstan/phpstan-symfony/pull/200#discussion_r724885819

So I'll go with the class_exists solution.

Does 
```
if (
			is_string($value)
			&& preg_match('/%env\((.*)\:.*\)%/U', $value, $matches) === 1
			&& strlen($matches[0]) === strlen($value)
			&& class_exists(EnvVarProcessor::class)
		) {
			$providedTypes = EnvVarProcessor::getProvidedTypes();

			return $this->typeStringResolver->resolve($providedTypes[$matches[1]] ?? 'bool|int|float|string|array');
		}
```
wouldn't be enough ?